tmt/trying.py

```diff
@@ -32,85 +30,89 @@
 USER_PLAN_NAME = "/user/plan"
 
 
-class Action(enum.Enum):
-    """
-    Available actions and their keyboard shortcuts
-    """
-
-    TEST = "t", "rediscover tests and execute them again"
-    LOGIN = "l", "log into the guest for experimenting"
-    HOST = "h", "run command on the local host"
-    VERBOSE = "v", "set the desired level of verbosity"
-    DEBUG = "b", "choose a different debugging level"
-
-    DISCOVER = "d", "gather information about tests to be executed"
-    PREPARE = "p", "prepare the environment for testing"
-    EXECUTE = "e", "run tests using the specified executor"
-    REPORT = "r", "provide test results overview and send reports"
-    FINISH = "f", "perform the user defined finishing tasks"
-    CLEANUP = "c", "clean up guests and prune the workdir"
+class ActionInfo:
+    """Information about a registered action"""
 
-    KEEP = "k", "exit the session but keep the run for later use"
-    QUIT = "q", "clean up the run and quit the session"
+    def __init__(
+        self, commands: tuple[str, ...], help_text: str, func: Callable[..., Any]
+    ) -> None:
+        self.commands = commands
+        self.help_text = help_text
+        self.func = func
 
-    START_LOGIN = "-", "jump directly to login after start"
-    START_ASK = "-", "do nothing without first asking the user"
-    START_TEST = "-", "start directly with executing detected tests"
+    @property
+    def primary_command(self) -> str:
+        """Return the primary (first) command"""
+        return self.commands[0]
 
     @property
     def key(self) -> str:
-        """
-        Keyboard shortcut
-        """
+        """Return the keyboard shortcut (first character of first command)"""
+        return self.commands[0][0]
 
-        return self.value[0]
+    @property
+    def full_name(self) -> str:
+        """Return the full command name (longest command)"""
+        return max(self.commands, key=len)
 
     @property
-    def description(self) -> str:
-        """
-        Action description
-        """
+    def menu(self) -> str:
```

Comment:
Maybe `menu_item`, and it can also be a cached property.

Reply:
changed to menu item